### PR TITLE
Mgmt-8424: Day 2 ignition endpoint check customization

### PR DESCRIPTION
**Pull request by rollandf**
# Assisted Pull Request

## Description
Add support to pass to agent a custom ignition endpoint along with a CA and  auth token.
    
Also use configured pool name if exists.

## List all the issues related to this PR

- [ ] New Feature
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [x] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Assignees

/cc @eranco74 
/cc @avishayt 
/cc @danielerez 
/cc @filanov 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: internal/host/hostcommands/api_vip_connectivity_check_cmd.go [hunk 1]
```diff
@@ -32,16 +34,40 @@ func (c *apivipConnectivityCheckCmd) GetSteps(ctx context.Context, host *models.
...
-
-	apiURL := fmt.Sprintf("http://%s:22624/config/worker", addressPart)
+	var caCert *string = nil
+	poolName := "worker"
+	if host.MachineConfigPoolName != "" {
+		poolName = host.MachineConfigPoolName
+	}
+	ignitionEndpointUrl := fmt.Sprintf("http://%s:22624/config/%s", addressPart, poolName)
+	if cluster.IgnitionEndpoint != nil && cluster.IgnitionEndpoint.URL != nil {
+		caCert = cluster.IgnitionEndpoint.CaCertificate
```
Comment (eranco74) [inline internal/host/hostcommands/api_vip_connectivity_check_cmd.go:54]: `CaCertificate` might be nil as well

File: internal/host/hostcommands/api_vip_connectivity_check_cmd.go [hunk 2]
```diff
@@ -32,16 +34,40 @@ func (c *apivipConnectivityCheckCmd) GetSteps(ctx context.Context, host *models.
...
+		url, err := url.Parse(*cluster.IgnitionEndpoint.URL)
+		if err != nil {
+			return nil, err
+		}
+		url.Path = path.Join(url.Path, poolName)
+		ignitionEndpointUrl = url.String()
+	}
```
Comment (eranco74) [inline internal/host/hostcommands/api_vip_connectivity_check_cmd.go:61]: Consider extracting this logic to a separate function `getIgnitionURL`
* the caCert part doesn't have to be in it.

**Review by eranco74 — COMMENTED** [review]
Looks good.

File: internal/host/hostcommands/api_vip_connectivity_check_cmd.go (hunk 1 above)
Comment (rollandf) [reply internal/host/hostcommands/api_vip_connectivity_check_cmd.go]: Fixed

File: internal/host/hostcommands/api_vip_connectivity_check_cmd.go (hunk 2 above)
Comment (rollandf) [reply internal/host/hostcommands/api_vip_connectivity_check_cmd.go]: Fixed

**Comment by rollandf** [comment]
Agent side WIP PR : https://github.com/openshift/assisted-installer-agent/pull/261

**Review by eranco74 — COMMENTED** [review]
/lgtm

File: internal/host/hostcommands/api_vip_connectivity_check_cmd.go [hunk 3]
```diff
@@ -32,16 +34,29 @@ func (c *apivipConnectivityCheckCmd) GetSteps(ctx context.Context, host *models.
...
 	request := models.APIVipConnectivityRequest{
-		URL: &apiURL,
+		URL: &ignitionEndpointUrl,
 	}
+
+	if cluster.IgnitionEndpoint != nil && cluster.IgnitionEndpoint.CaCertificate != nil {
+		request.CaCertificate = cluster.IgnitionEndpoint.CaCertificate
+	}
+
+	if commonHost.IgnitionEndpointToken != "" {
```
Comment (eranco74) [inline internal/host/hostcommands/api_vip_connectivity_check_cmd.go:56]: :+1:

**Pull request merged**